internal/controllers/provisioningrequest_controller.go

```diff
@@ -284,6 +284,24 @@ func (t *provisioningRequestReconcilerTask) run(ctx context.Context) (ctrl.Resul
 		if !utils.IsClusterProvisionCompleted(t.object) || requeue {
 			return requeueWithLongInterval(), nil
 		}
+
+		shouldUpgrade, err := t.IsUpgradeRequested(ctx, renderedClusterInstance.GetName())
+		fmt.Println(shouldUpgrade)
+		if err != nil {
+			return requeueWithError(err)
+		}
+		if utils.IsClusterProvisionCompleted(t.object) && shouldUpgrade {
```

Comment:
have you done e2e test including finalize? After post pivot, we would be getting the new OCP version. handleUpgrade would be getting skipped. This condition probably should be more like "shouldUpgrade || upgradeInProgress"?

Reply:
not with finalize.
The condition will change after Angie's PR is merged and with the next story.
Probably as per document we will add a new condition `ClusterUpgraded` and we can check with that.